data-prepper-plugins/opensearch-api-source/src/test/java/org/opensearch/dataprepper/plugins/source/opensearchapi/OpenSearchAPISourceTest.java

```diff
@@ -0,0 +1,623 @@
+        lenient().when(sourceConfig.getThreadCount()).thenReturn(DEFAULT_THREAD_COUNT);
+        lenient().when(sourceConfig.getMaxConnectionCount()).thenReturn(MAX_CONNECTIONS_COUNT);
+        lenient().when(sourceConfig.getMaxPendingRequests()).thenReturn(MAX_PENDING_REQUESTS_COUNT);
+        lenient().when(sourceConfig.hasHealthCheckService()).thenReturn(true);
+        lenient().when(sourceConfig.getCompression()).thenReturn(CompressionOption.NONE);
+
+        MetricsTestUtil.initMetrics();
+        pluginMetrics = PluginMetrics.fromNames(PLUGIN_NAME, TEST_PIPELINE_NAME);
+
+        pluginFactory = mock(PluginFactory.class);
+        final ArmeriaHttpAuthenticationProvider authenticationProvider = new HttpBasicArmeriaHttpAuthenticationProvider(new HttpBasicAuthenticationConfig("test", "test"));
+        when(pluginFactory.loadPlugin(eq(ArmeriaHttpAuthenticationProvider.class), any(PluginSetting.class)))
+                .thenReturn(authenticationProvider);
+
+        testBuffer = getBuffer();
+        pipelineDescription = mock(PipelineDescription.class);
+        when(pipelineDescription.getPipelineName()).thenReturn(TEST_PIPELINE_NAME);
+
+        openSearchAPISource = new OpenSearchAPISource(sourceConfig, pluginMetrics, pluginFactory, pipelineDescription);
+    }
+
+    @AfterEach
+    public void cleanUp() {
+        if (openSearchAPISource != null) {
+            openSearchAPISource.stop();
+        }
+    }
+
+    private void assertSecureResponseWithStatusCode(final AggregatedHttpResponse response, final HttpStatus expectedStatus) {
+        assertThat("Http Status", response.status(), equalTo(expectedStatus));
+
+        final List<String> headerKeys = response.headers()
+                .stream()
+                .map(Map.Entry::getKey)
+                .map(AsciiString::toString)
+                .collect(Collectors.toList());
+        assertThat("Response Header Keys", headerKeys, not(contains("server")));
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    public void testBulkRequestAPIResponse200(boolean includeIndexInPath) throws IOException {
+        int numberOfRecords = 1;
+        testBulkRequestAPI200(includeIndexInPath, false, numberOfRecords);
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    public void testBulkRequestAPICompressionResponse200(boolean includeIndexInPath) throws IOException {
+        int numberOfRecords = 1;
+        testBulkRequestAPI200(includeIndexInPath, true, numberOfRecords);
+    }
+
+    @Test
+    public void testHealthCheck() {
+        // Prepare
+        openSearchAPISource.start(testBuffer);
+
+        // When
+        WebClient.of().execute(RequestHeaders.builder()
+                        .scheme(SessionProtocol.HTTP)
+                        .authority(AUTHORITY)
+                        .method(HttpMethod.GET)
+                        .path("/health")
```

Comment:
Do we want a health check endpoint? This is not something that the OpenSearch API would normally have.

If anything, maybe we should support the root API?

```
GET /
```

We could add this in a follow-on PR. But, I think we want to remove the `/health` endpoint.

Do you want to create a follow-on PR for the root endpoint?

Reply:
Thanks @dlvenable. I will resolve this. The default health check endpoint is `/health` which is the default endpoint for Http source. I will modify the BastHttpSource implementation to ensure that the OpenSearch API source can control the endpoint `/` for health check.